### PR TITLE
GEOMESA-1936,GEOMESA-2427 Support non-point geometries and Booleans in Parquet

### PR DESCRIPTION
**Pull request by elahrvivaz**
* Support linestrings, multipoints, polygons, multilinestrings, multipolygons
* Move ObjectTypes.JSON to secondary typing (primary now STRING)
* Make parquet FilterConverter an object, run in a single pass

Signed-off-by: Emilio Lahr-Vivaz <elahrvivaz@ccri.com>

### CODE REVIEW

**Comment by elahrvivaz** [comment]
note: stacks on top of #2260

File: docs/user/geojson.rst [hunk 1]
```diff
@@ -170,9 +170,8 @@ REST Access
 ^^^^^^^^^^^
 
 The ``GeoJsonIndex`` is also exposed through a REST endpoint. Currently, the REST endpoint does not support
-updates or deletes of existing features, or transformation of resposes. Furthermore, it requires Accumulo as
-the backing data store. It may be installed in GeoServer by extracting the install file into
-``geoserver/WEB-INF/lib``:
+transformation of responses. Furthermore, it requires Accumulo as the backing data store. It may be installed
```
Comment (jnh5y) [inline docs/user/geojson.rst]: Why does the GeoJSON require Accumulo?

File: geomesa-fs/geomesa-fs-storage/geomesa-fs-storage-parquet/src/main/scala/org/locationtech/geomesa/parquet/ParquetFileSystemStorage.scala [hunk 2]
```diff
@@ -44,39 +48,59 @@ class ParquetFileSystemStorage(context: FileSystemContext, metadata: StorageMeta
...
-        val builder = new SimpleFeatureTypeBuilder()
-        builder.init(tsft)
-        filters.foreach(f => builder.add(metadata.sft.getDescriptor(f)))
-        val psft = builder.buildFeatureType()
-        psft.getUserData.putAll(metadata.sft.getUserData)
-        psft
-      }
-    }.getOrElse(metadata.sft)
-
-    // TODO GEOMESA-1954 move this filter conversion higher up in the chain
```
Comment (jnh5y) [inline geomesa-fs/geomesa-fs-storage/geomesa-fs-storage-parquet/src/main/scala/org/locationtech/geomesa/parquet/ParquetFileSystemStorage.scala]: Does this PR address this ticket?  We closed it as "won't do".

File: docs/user/geojson.rst (hunk 1 above)
Comment (elahrvivaz) [reply docs/user/geojson.rst]: It works with any store, but we only made a tar.gz bundle for Accumulo.

File: geomesa-fs/geomesa-fs-storage/geomesa-fs-storage-parquet/src/main/scala/org/locationtech/geomesa/parquet/ParquetFileSystemStorage.scala (hunk 2 above)
Comment (elahrvivaz) [reply geomesa-fs/geomesa-fs-storage/geomesa-fs-storage-parquet/src/main/scala/org/locationtech/geomesa/parquet/ParquetFileSystemStorage.scala]: yeah i think i closed it b/c we decided there wasn't really any reason to do it, so i just removed the TODO here.

**Pull request merged**